packages/flutter/lib/src/material/carousel.dart

```diff
@@ -140,16 +201,40 @@ class CarouselView extends StatefulWidget {
   /// Defaults to false.
   final bool reverse;
 
+
+  /// Whether the "squished" item is allowed to expand to the max size.
+  ///
+  /// If this is false, the layout of the carousel doesn't change. This is especially
+  /// useful when a weight list in [CarouselView.weighted] has a max item in the
+  /// middle and at least one small item on either side, such as `[1, 7, 1]`,
+  /// the first or the last item cannot expand to the max size. If this is true,
+  /// there will be some space before the first item or after the last item
+  /// coming so every items have a chance to be fully expanded.
+  ///
+  /// Defaults to true.
+  final bool allowFullyExpand;
```

Comment:
This name is not great :). Would something like `allowFirstAndLastToExpand` be better? Maybe other reviewers have better suggestions.

Reply:
Ah I wondered the same above. I thought maybe `consumeMaxWeight`?

Reply:
This one is shorter! I'll update it! But please let me know if there're any other preference😃